### ISSUE TITLE
Named fragments don't work with mutations

### ISSUE THREAD

**Issue by Poincare** [apollostack/apollo-client#273]
For example, something like this won't work correctly:

```
mutation {
  createAuthor(firstName: "John", lastName: "Smith") {
    ...authorDetails
  }
  fragment authorDetails on Author {
    firstName
    lastName
  }
}
```
because the `mutate` method within `QueryManager` works with the mutation `OperationDefinition` rather than the `Document` containing the mutation. Should be an easy fix - see PR #251 since that's pretty much the same thing but for queries.

**Comment by helfer** [comment]
good catch!

**Comment by Poincare** [comment]
Fixed w/ PR #274.
